internal/alloycli/cmd_validate.go

```diff
@@ -0,0 +1,91 @@
+package alloycli
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/internal/service"
+	"github.com/grafana/alloy/internal/service/cluster"
+	"github.com/grafana/alloy/internal/service/http"
+	"github.com/grafana/alloy/internal/service/labelstore"
+	"github.com/grafana/alloy/internal/service/otel"
+	"github.com/grafana/alloy/internal/service/remotecfg"
+	"github.com/grafana/alloy/internal/service/ui"
+	"github.com/grafana/alloy/internal/validator"
+	"github.com/spf13/cobra"
+)
+
+func validateCommand() *cobra.Command {
+	v := &alloyValidate{
+		configFormat: "alloy",
+		minStability: featuregate.StabilityGenerallyAvailable,
+	}
+
+	cmd := &cobra.Command{
+		Use:          "validate [flags] file",
+		Short:        "Validate a configuration file",
+		Long:         ``,
+		Args:         cobra.RangeArgs(0, 1),
```

Comment:
I don't think we documented that there can be 0 arguments?

Reply:
Yeah it should not be possible to have 0 args. I think I based the command from `fmt` first where you could pass config from stdin. Will fix that